libraries/AP_RangeFinder/AP_RangeFinder_UPFLOW_TOF.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "AP_RangeFinder_config.h"
+#include "../AP_OpticalFlow/AP_OpticalFlow_UPFLOW_Tx.h"
+
+#if AP_RANGEFINDER_UPFLOW_Tx_ENABLED
+
+#include "AP_RangeFinder.h"
+#include "AP_RangeFinder_Backend.h"
+#include "AP_RangeFinder_Params.h"
+#include "GCS_MAVLink/GCS_MAVLink.h"
+
+class AP_RangeFinder_UPFLOW_TOF : public AP_RangeFinder_Backend
+{
+public:
+    // constructor
+    AP_RangeFinder_UPFLOW_TOF(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params);
+
+    // static detection function
+    static bool detect();
+
+    // update state
+    void update(void) override;
+
+protected:
+
+    MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const override {
+        return MAV_DISTANCE_SENSOR_ULTRASOUND;
+    }
+
+private:
+
+    uint8_t glitch_count{0};          // glitch counter
```

Comment:
```suggestion
    uint8_t glitch_count;          // glitch counter
```

We zero memory except that which is allocated on the stack - saves a lot of flash, but does mean you have to get in the habit of not reflexively zeroing things :-)